src/Microsoft.ML.TimeSeries/DTRootCauseLocalization.cs

```diff
@@ -0,0 +1,305 @@
+        internal const string UserName = "DT Root Cause Localization Transform";
+        internal const string LoaderSignature = "DTRootCauseLTransform";
+
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "DTRCL",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(DTRootCauseLocalizationTransformer).Assembly.FullName);
+        }
+
+        private const string RegistrationName = "RootCauseLocalization";
+
+        internal sealed class Column : OneToOneColumn
+        {
+            internal static Column Parse(string str)
+            {
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            internal bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        internal class Options : TransformInputBase
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition(s) (optional form: name:src)", Name = "Column", ShortName = "col", SortOrder = 1)]
+            public Column[] Columns;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Weight for getting the score for the root cause item.", ShortName = "Beta", SortOrder = 2)]
+            public double Beta = DTRootCauseLocalizationEstimator.Defaults.Beta;
+
+        }
+
+        /// <summary>
+        /// The input and output column pairs passed to this <see cref="ITransformer"/>.
+        /// </summary>
+        internal IReadOnlyCollection<(string outputColumnName, string inputColumnName)> Columns => ColumnPairs.AsReadOnly();
+
+        private readonly double _beta;
+
+        /// <summary>
+        /// Localization root cause for multi-dimensional anomaly.
+        /// </summary>
+        /// <param name="env">The estimator's local <see cref="IHostEnvironment"/>.</param>
+        /// <param name="beta">Weight for generating score.</param>
+        /// <param name="columns">The name of the columns (first item of the tuple), and the name of the resulting output column (second item of the tuple).</param>
+
+        internal DTRootCauseLocalizationTransformer(IHostEnvironment env, double beta = DTRootCauseLocalizationEstimator.Defaults.Beta, params (string outputColumnName, string inputColumnName)[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(RegistrationName), columns)
+        {
+            Host.CheckUserArg(beta >= 0 && beta <= 1, nameof(Options.Beta), "Must be in [0,1]");
+
+            _beta = beta;
+        }
+
+        // Factory method for SignatureDataTransform.
```

Comment:
>SignatureDataTransform [](start = 30, length = 22)

This (and the matching attribute) is only needed if you want to use this component from the command line. If this is the intention, please add a test for running this using maml.

Reply:
I have no idea about the maml, could you help provide more information?

Reply:
> > SignatureDataTransform [](start = 30, length = 22)
> 
> This (and the matching attribute) is only needed if you want to use this component from the command line. If this is the intention, please add a test for running this using maml.

I have no idea about the maml, could you help provide more information?

Reply:
> > > SignatureDataTransform [](start = 30, length = 22)
> > 
> > 
> > This (and the matching attribute) is only needed if you want to use this component from the command line. If this is the intention, please add a test for running this using maml.
> 
> I have no idea about the maml, could you help provide more information?

maml is not needed and I have removed this method